File: auth-service/src/main/java/com/eirsteir/coffeewithme/authservice/repository/AccountRepository.java
package com.eirsteir.coffeewithme.authservice.repository;

import com.eirsteir.coffeewithme.authservice.domain.Account;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface AccountRepository extends JpaRepository<Account, Long> {

  Optional<Account> findByEmail(String email);
}

File: auth-service/src/main/kotlin/com/eirsteir/coffeewithme/authservice/repository/AccountRepository.kt
package com.eirsteir.coffeewithme.authservice.repository

import com.eirsteir.coffeewithme.authservice.domain.Account
import org.springframework.data.jpa.repository.JpaRepository

interface AccountRepository : JpaRepository<Account, Long> {
    fun findByEmail(email: String): Account?
}